skyportal/handlers/api/source.py

```diff
@@ -60,18 +69,56 @@ def get(self, source_id=None):
                          .joinedload(Photometry.instrument)
                          .joinedload(Instrument.telescope)])
         elif page_number:
-            page = int(page_number)
+            try:
+                page = int(page_number)
+            except ValueError:
+                return self.error("Invalid page number value.")
             q = Source.query.filter(Source.id.in_(DBSession.query(
                 GroupSource.source_id).filter(GroupSource.group_id.in_(
                     [g.id for g in self.current_user.groups]))))
-            info['totalMatches'] = q.count()
+            if sourceID:
+                q = q.filter(Source.id.contains(sourceID.strip()))
+            if ra and dec and radius:
```

Comment:
Silent failure if not all three specified?

Reply:
Do you think if they provide any of these but not all three we should return an error message? Now it just omits those params from the query.

Reply:
Right, because it shows intent to do spatial filtering, just that it is done incorrectly.